plugins/bulk-import/src/components/Repositories/RepositoriesList.tsx

```diff
@@ -1,48 +1,49 @@
 import React from 'react';
 import { useLocation, useNavigate } from 'react-router-dom';
 
-import { ErrorPage, Table } from '@backstage/core-components';
+import { ErrorPage } from '@backstage/core-components';
 
-import { makeStyles } from '@material-ui/core';
+import {
+  Box,
+  Paper,
+  Table,
+  TableContainer,
+  TablePagination,
+} from '@material-ui/core';
 
 import { useDeleteDialog, useDrawer } from '@janus-idp/shared-react';
 
 import { useAddedRepositories } from '../../hooks/useAddedRepositories';
-import { AddRepositoryData } from '../../types';
+import { AddRepositoryData, Order } from '../../types';
+import { getComparator } from '../../utils/repository-utils';
+import { RepositoriesHeader } from '../AddRepositories/RepositoriesHeader';
+import { AddedRepositoriesTableBody } from './AddedRepositoriesTableBody';
 import DeleteRepositoryDialog from './DeleteRepositoryDialog';
 import EditCatalogInfo from './EditCatalogInfo';
-import { columns } from './RepositoriesListColumns';
 import { RepositoriesListToolbar } from './RepositoriesListToolbar';
 
-const useStyles = makeStyles(theme => ({
-  empty: {
-    padding: theme.spacing(2),
-    display: 'flex',
-    justifyContent: 'center',
-  },
-}));
-
 export const RepositoriesList = () => {
   const navigate = useNavigate();
   const location = useLocation();
   const searchParams = new URLSearchParams(location.search);
+  const [order, setOrder] = React.useState<Order>('asc');
+  const [orderBy, setOrderBy] = React.useState<string>();
   const { openDialog, setOpenDialog, deleteComponent } = useDeleteDialog();
   const { openDrawer, setOpenDrawer, drawerData } = useDrawer();
   const [pageNumber, setPageNumber] = React.useState(0);
   const [rowsPerPage, setRowsPerPage] = React.useState(5);
-  const [searchString, setSearchString] = React.useState('');
-  const classes = useStyles();
+  const [debouncedSearch, setDebouncedSearch] = React.useState('');
```

Comment:
Imo: It would be nice to replace this with `useQueryParamState` from `@ackstage/core-components`. It has an optional debounceTime as well.

https://backstage.io/docs/reference/core-components.usequeryparamstate/

Reply:
As of today, we do not add the search text in the query param of the bulk-import URL
I can create a separate ticket to address this